sql/sql_yacc.yy

```diff
@@ -7723,11 +7723,13 @@ btree_or_rtree:
 key_list:
           key_list ',' key_part order_dir
           {
-            Lex->last_key->columns.push_back($3, thd->mem_root);
+            if ($3 != NULL)
```

Comment:
As far as i see, it couldn't be null here. Parsing aborts immediately before it reaches here in `key_part`.
So it's reasonable to add DBUG_ASSERT instead.